docs/architecture/adr-028-public-key-addresses.md

```diff
@@ -56,113 +97,145 @@ The current multisig public keys use amino serialization to generate the address
 those public keys and their address formatting, and call them "legacy amino" multisig public keys
 in protobuf. We will also create multisig public keys without amino addresses to be described below.
 
+### Hash Function Choice
 
-### Canonical Address Format
+As in other parts of the Cosmos SDK, we will use `sha256`.
 
-We have three types of accounts we would like to create addresses for in the future:
-- regular public key addresses for new signature algorithms (ex. `sr25519`).
-- public key addresses for multisig public keys that don't use amino encoding
-- module accounts: basically any accounts which cannot sign transactions and
-which are managed internally by modules
+### Simple Account Address
 
-To address all of these use cases we propose the following basic `AddressHash` function,
-based on the discussions in [\#5694](https://github.com/cosmos/cosmos-sdk/issues/5694):
+We start with defining a hash base algorithm for generating addresses. Notably, it's used for accounts represented by a single key pair. For each public key schema we have to have an associated `typ` string, which we are discussing in a section below. `hash` is a cryptographic hash function defined in the previous section.
 
 ```go
-func AddressHash(prefix string, contents []byte) []byte {
-	preImage := []byte(prefix)
-	if len(contents) != 0 {
-		preImage = append(preImage, 0)
-		preImage = append(preImage, contents...)
-	}
-	return sha256.Sum256(preImage)[:20]
+const A_LEN = 32
```

Comment:
What's not clear to me is if `A_LEN` is constant for all simple account addresses, or if it depends on `typ`. I can imagine introducing in the future some pk schemes with different public key sizes.

Reply:
It doesn't depend on `typ`. `A_LEN` is a length of all `Hash` based addresses.